src/app/api/v1/user/email/route.ts

```diff
@@ -0,0 +1,142 @@
+
+export async function POST(req: NextRequest) {
+  const token = await getToken({ req });
+
+  if (token) {
+    try {
+      const body: EmailAddRequest = await req.json();
+      const subscriber = await getSubscriberByEmail(token.email);
+      const emailCount = 1 + (subscriber.email_addresses?.length ?? 0); // primary + verified + unverified emails
+      const validatedEmail = validateEmailAddress(body.email);
+
+      if (validatedEmail === null) {
+        return NextResponse.json({
+          success: false,
+          status: 400,
+          message: l10n.getString("user-add-invalid-email"),
+        });
+      }
+
+      if (emailCount >= AppConstants.MAX_NUM_ADDRESSES) {
+        return NextResponse.json({
+          success: false,
+          status: 400,
+          message: l10n.getString("user-add-too-many-emails"),
+        });
+      }
+
+      // checkForDuplicateEmail
+      const emailLowerCase = validatedEmail.email.toLowerCase();
+      if (emailLowerCase === subscriber.primary_email.toLowerCase()) {
+        return NextResponse.json({
+          success: false,
+          status: 400,
+          message: l10n.getString("user-add-duplicate-email"),
+        });
+      }
+
+      for (const secondaryEmail of subscriber.email_addresses) {
+        if (emailLowerCase === secondaryEmail.email.toLowerCase()) {
+          return NextResponse.json({
+            success: false,
+            status: 400,
+            message: l10n.getString("user-add-duplicate-email"),
+          });
+        }
+      }
+
+      const unverifiedSubscriber = await addSubscriberUnverifiedEmailHash(
+        subscriber,
+        validatedEmail.email
+      );
+
+      await sendVerificationEmail(subscriber, unverifiedSubscriber.id);
+
+      return NextResponse.json({
+        success: true,
+        newEmailCount: emailCount + 1,
+        message: "Sent the verification email",
+      });
+      return NextResponse.json("ok");
+    } catch (e) {
+      console.error(e);
+      if (e.message === "error-email-validation-pending") {
+        return NextResponse.json(
+          {
+            success: false,
+            message: "Verification email recently sent, try again later",
+          },
+          { status: 429 }
+        );
+      }
+      return NextResponse.json({ success: false }, { status: 500 });
+    }
+  } else {
+    // Not Signed in, redirect to home
+    return NextResponse.redirect(AppConstants.SERVER_URL, 301);
+  }
+}
+
+interface EmailDeleteRequest {
+  emailId: string;
+}
+
+export async function DELETE(req: NextRequest) {
+  const token = await getToken({ req });
+
+  if (token) {
+    try {
+      const { emailId }: EmailDeleteRequest = await req.json();
+      const subscriber = await getSubscriberByEmail(token.email);
+      const existingEmail = await getEmailById(emailId);
+
+      if (existingEmail?.subscriber_id !== subscriber.id) {
+        return NextResponse.json({
+          success: false,
+          status: 400,
```

Comment:
Same Q here re: `{ status: 400 }` as a second arg to `NextResponse.json({…}, { status: 400 })`.